### PR TITLE
Update `nexus_upload` action to accept a nexus mount_path config item

### PR DESCRIPTION
**Pull request by mfurtak**
Squashed and lightly adjusted from @chrisgibbs #1305

### CODE REVIEW

File: spec/actions_specs/nexus_upload_spec.rb [hunk 1]
```diff
@@ -7,7 +7,12 @@
       end
 
       it "upload url is set correctly" do
-        expect(Fastlane::Actions::NexusUploadAction.upload_url(endpoint: "http://localhost:8081")).to eq "http://localhost:8081/nexus/service/local/artifact/maven/content"
+        expect(Fastlane::Actions::NexusUploadAction.upload_url(endpoint: "http://localhost:8081",
+          mount_path: "/nexus")).to eq "http://localhost:8081/nexus/service/local/artifact/maven/content"
+        expect(Fastlane::Actions::NexusUploadAction.upload_url(endpoint: "http://localhost:8081",
+          mount_path: "/custom-nexus")).to eq "http://localhost:8081/custom-nexus/service/local/artifact/maven/content"
+        expect(Fastlane::Actions::NexusUploadAction.upload_url(endpoint: "http://localhost:8081",
+          mount_path: "")).to eq "http://localhost:8081/service/local/artifact/maven/content"
```
Comment (i2amsam) [inline spec/actions_specs/nexus_upload_spec.rb]: Add a test for the backward-compatible case

Comment (i2amsam) [inline spec/actions_specs/nexus_upload_spec.rb]: Oh, I see, this looks good.

**Comment by i2amsam** [comment]
:+1: looks great!

**Comment by chrisgibbs** [comment]
I got my PR squashed but your upload_url is more DRY. Should I update my PR or just decline it in favor of this?

**Comment by mfurtak** [comment]
Hi @chrisgibbs - sorry for the confusion. I just figured I'd save you the effort since I had the time... I'll get this one in right now and close the other.

Thanks again for your contribution! :tada:

**Pull request merged**